pkgs/os-specific/linux/zmalllinugz/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchurl, coreutils, flex, bison, libelf, bc, hostname, perl }:
+
+
+let  
+  version = "5.3.8";
+  src = fetchurl {
+    url = "mirror://kernel/linux/kernel/v5.x/linux-${version}.tar.xz";
+    sha256 = "0jb6yya9yx4z52p5m32dqj0kgc6aaz9df8mvq0hzy40bqb3czwvq";
+  };
+  config = ./zmalllinugz.config;
+
+in
+stdenv.mkDerivation rec {
+  name = "zmalllinugz";
+
+  srcs = [ src config ];
+
+  dontConfigure=true;
+  buildInputs = [ coreutils flex bison libelf bc hostname perl];
+  
+  unpackPhase = ''
+    array=($srcs)
```

Comment:
Are not all sources unpacked by default? Could you do the additional steps in the `postUnpack` as to avoid overriding the phase?

Reply:
not sure what is the best to add tar.gz as source + the config. By default default unpackPhase will try to unpack the .config and fails. Any suggestion?

Reply:
BTW: I see lot's of heuristics in the builder script. The unpackPhase, as mentioned tries to unpack several sources, but expects only one sourceDir. With the same logic, the unpackPhase could just ignore sources that do not have a known extension and pass them as they are to the next phase. Maybe I should create an issue for this. Or am I missing something? I would love to understand better this, as the documentation is bit poor in this sense. (about semantics, intentions etc.)